### PR TITLE
qemu.test: Update multi_disk related cases

### PR DESCRIPTION
**Pull request by XueqiangWei**
(1) add case with_virtio_scsi_TMF
(2) add windows support for case multi_lun
(3) add windows support for case multi_scsiid_lun
(4) add windows support for case multi_bus_scsiid_lun

ID: 1572480 1321731
Signed-off-by: Xueqiang Wei <xuwei@redhat.com>

### CODE REVIEW

**Comment by XueqiangWei** [comment]
depends on avocado-framework/avocado-vt#1556

**Comment by XueqiangWei** [comment]
(1/1) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.5.x86_64.io-github-autotest-qemu.multi_disk.with_virtio_scsi_TMF: PASS (1701.46 s)

**Comment by peixiu** [comment]
Hi xuwei,
For multi_disks case, suggest add reboot/shutdown check after disk number or io operation finished.
I hit a issue today, guest shutdown or reboot will bsod with 256 disks attached.

Bug 1589046 - [virtio-win][vioscsi]reboot/shutdown guest hit bsod(9F) with 256 disks attached --win2008-32

File: qemu/tests/multi_disk.py [hunk 1]
```diff
@@ -221,15 +200,16 @@ def _get_disk_index(session, image_size, disk_indexs):
     error_context.context("Start the guest with those disks", logging.info)
     vm = env.get_vm(params["main_vm"])
     vm.create(timeout=max(10, stg_image_num), params=params)
-    session = vm.wait_for_login(timeout=int(params.get("login_timeout", 360)))
+    login_timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=login_timeout)
 
     n_repeat = int(params.get("n_repeat", "1"))
     file_system = [_.strip() for _ in params.get("file_system").split()]
```
Comment (hereischen) [inline qemu/tests/multi_disk.py]: Please use `params["file_system"]` to indicate it is a required param.

**Comment by XueqiangWei** [comment]
depends on avocado-framework/avocado-vt#1640

**Comment by XueqiangWei** [comment]
(01/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.i
o-github-autotest-qemu.multi_disk.one_disk_repeat: PASS (100.04 s)
 (02/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.i
o-github-autotest-qemu.multi_disk.max_disk: PASS (202.41 s)
 (03/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.i
o-github-autotest-qemu.multi_disk.all_drive_format_types: PASS (62.31 s)
 (04/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.i
o-github-autotest-qemu.multi_disk.virtio_scsi_variants.block: PASS (100.97 s)
 (05/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.i
o-github-autotest-qemu.multi_disk.virtio_scsi_variants.generic: PASS (102.54 s)
 (06/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.i
o-github-autotest-qemu.multi_disk.virtio_scsi_variants.multi_lun: PASS (1162.07 
s)
 (07/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.i
o-github-autotest-qemu.multi_disk.virtio_scsi_variants.multi_scsiid_lun: PASS (3
691.79 s)
 (08/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.i
o-github-autotest-qemu.multi_disk.virtio_scsi_variants.multi_bus_scsiid_lun: PASS (571.94 s)
(09/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.multi_disk.one_disk_repeat: PASS (238.92 s)
 (10/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.multi_disk.max_disk: PASS (638.10 s)
 (11/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.multi_disk.all_drive_format_types: PASS (188.34 s)
 (12/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.multi_disk.virtio_scsi_variants.block: PASS (277.96 s)
 (13/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.multi_disk.virtio_scsi_variants.generic: PASS (231.28 s)
 (14/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.multi_disk.virtio_scsi_variants.multi_lun: PASS (1502.97 s)
 (15/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.multi_disk.virtio_scsi_variants.multi_scsiid_lun: PASS (4205.35 s)
 (16/16) Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.multi_disk.virtio_scsi_variants.multi_bus_scsiid_lun: PASS (1018.45 s)
(1/1) Host_RHEL.m7.u5.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.5.x86_64.io-github-autotest-qemu.multi_disk.with_virtio_scsi_TMF: PASS (1623.08 s)

**Comment by XueqiangWei** [comment]
@hereischen @aliang123 @YongxueHong @lolyu Please help review it, many thanks.

File: qemu/tests/cfg/multi_disk.cfg [hunk 2]
```diff
@@ -45,6 +53,15 @@
                     image_size_image1 = 20G
                 Windows:
                     image_size_image1 = 30G
+        - with_virtio_scsi_TMF:
+            only Linux
+            only virtio_blk
```
[the remark this answers is not in the document]
Comment (XueqiangWei) [reply qemu/tests/cfg/multi_disk.cfg]: Just test virtio_blk is ok. The scenario is \"one virtio_blk as system disk and 300 virtio-scsi disks as data disk\".

[the remark this answers is not in the document]
Comment (ngu-niny) [reply qemu/tests/cfg/multi_disk.cfg]: @YongxueHong Checked with Qunfang, we should support spapr_vscsi.

[the remark this answers is not in the document]
Comment (YongxueHong) [reply qemu/tests/cfg/multi_disk.cfg]: @ngu-niny this patch only test for virtio_blk and virtio_scsi interface now, so you could add a new RFE requirement for spapr_vscsi if you need, than I will develop it.

[the remark this answers is not in the document]
Comment (ngu-niny) [reply qemu/tests/cfg/multi_disk.cfg]: Ok, on problem.

File: qemu/tests/cfg/multi_disk.cfg [hunk 3]
```diff
@@ -69,18 +86,22 @@
             # virtio-scsi driver support from RHEL6.3
             no RHEL.3 RHEL.4 RHEL.5 RHEL.6.0 RHEL.6.1 RHEL.6.2
             # Decrease length of the command
-            stg_image_size = 40M
```
[the remark this answers is not in the document]
Comment (ngu-niny) [reply qemu/tests/cfg/multi_disk.cfg]: We choose to use xhci as the 1st choice.

[the remark this answers is not in the document]
Comment (ngu-niny) [reply qemu/tests/cfg/multi_disk.cfg]: I meant for ppc.

File: qemu/tests/cfg/multi_disk.cfg [hunk 4]
```diff
@@ -45,6 +53,15 @@
                     image_size_image1 = 20G
                 Windows:
                     image_size_image1 = 30G
+        - with_virtio_scsi_TMF:
+            only Linux
+            only virtio_blk
+            #drive_format_image1 = virtio
```
[the remark this answers is not in the document]
Comment (XueqiangWei) [reply qemu/tests/cfg/multi_disk.cfg]: Removed.

File: qemu/tests/cfg/multi_disk.cfg (hunk 3 above)
[the remark this answers is not in the document]
Comment (XueqiangWei) [reply qemu/tests/cfg/multi_disk.cfg]: updated, thanks

**Comment by peixiu** [comment]
Thanks xuwei, looks good to me~
Acked by: Peixiu phou@redhat.com

**Pull request merged**